src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,60 @@
+package duke;
+
+import duke.exception.LoadFileFailDukeException;
+import duke.exception.WriteFileFailDukeException;
+import place.Place;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class Storage {
+    private String filePath;
+
+    /**
+     * Initialise filePath for continuous referencing.
+     *
+     * @param filePath filePath to save file
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+
+    /**
+     * Try to load raw data from filePath and deserialize everything.
+     *
+     * @throws LoadFileFailDukeException if file or raw data can't be loaded
+     */
+    @SuppressWarnings("unchecked")
```

Comment:
Perhaps you can try to resolve the warning instead of suppressing the warning as it might lead to more underlying bugs.  Is there a reason for this decision? 😄

Reply:
👍 I tried searching for an appropriate way to handle reading Objects. Unfortunately, all solutions I have come across so far has no "clean" way of dealing with Object casting as there is a implied order of processing.